### PR TITLE
feat: implement configreloaded event

### PR DESCRIPTION
**Pull request by diniamo**
@yavko rebase on master after you merge this please

### CODE REVIEW

**Comment by yavko** [comment]
I'll wait for #208 to get merged first, since changing implementations to the event listener stuff

**Comment by diniamo** [comment]
I already did that for my project: https://github.com/diniamo/hyprland-rs/tree/configreloaded-and-workspacerules

**Comment by yavko** [comment]
> ~~I already did that for my project: https://github.com/diniamo/hyprland-rs/tree/configreloaded-and-workspacerules~~
> 
> Never mind, that was before the big merge, but I'm not gonna rebase master, I'm gonna rebase add-events once you rebase that.

Okay one sec

File: src/shared.rs [hunk 1]
```diff
@@ -19,6 +20,8 @@ pub enum HyprError {
     FromUtf8Error(std::string::FromUtf8Error),
     /// Dispatcher returned non `ok` value
     NotOkDispatch(String),
+    /// Failed to parse an integer
+    ParseError(<i32 as FromStr>::Err),
```
Comment (yavko) [inline src/shared.rs:24]: Please instead create an error using the new Hyprland-rs error types that are more detailed, and can be used to provide more insight

Comment (yavko) [reply src/shared.rs:24]: also, even if you did use this, you should've used https://doc.rust-lang.org/std/num/struct.ParseIntError.html

File: src/event_listener/shared.rs [hunk 2]
```diff
@@ -563,6 +565,11 @@ pub struct WindowFloatEventData {
 unsafe impl Send for WindowFloatEventData {}
 #[allow(unsafe_code)]
 unsafe impl Sync for WindowFloatEventData {}
+
+/// This struct is for events that have no data
+#[derive(Debug, Clone, Copy)]
+pub struct EmptyEventData;
```
Comment (yavko) [inline src/event_listener/shared.rs:571]: are you 100% sure you can't figure out how to do it without?

**Review by yavko — CHANGES_REQUESTED** [review]

**Comment by yavko** [comment]
also fix the merge conflicts

**Pull request merged**